### PR TITLE
Add Registry Credentials to Tests

### PR DESCRIPTION
**Pull request by nathanjsweet**
In order to get around image registry pull limits,
credentials can be set.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

### CODE REVIEW

File: test/k8sT/Bandwidth.go [hunk 1]
```diff
@@ -51,6 +52,8 @@ var _ = Describe("K8sBandwidthTest", func() {
 
 	BeforeAll(func() {
 		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		err := kubectl.AddRegistryCredentials(config.CiliumTestConfig.RegistryCredentials, config.RegistryDomain)
```
Comment (nebril) [inline test/k8sT/Bandwidth.go]: I think this should be moved to either `helpers.CreateKubectl` or `kubectl.PrepareCluster`, I don't think there is need for it to be in every test suite.

**Review by nebril — CHANGES_REQUESTED** [review]
I think that's a great change, I have one comment left inline.

**Comment by nathanjsweet** [comment]
test-me-please

File: test/config/config.go [hunk 2]
```diff
@@ -90,6 +96,8 @@ func (c *CiliumTestConfigType) ParseFlags() {
 		"Specifies timeout for test run")
 	flagset.StringVar(&c.Kubeconfig, "cilium.kubeconfig", "",
 		"Kubeconfig to be used for k8s tests")
+	flagset.StringVar(&c.RegistryCredentials, "cilium.registryCredentials", "",
```
Comment (christarazi) [inline test/config/config.go:99]: nit: I think there's a precedence for kebab case (`registry-credentials`)

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -401,6 +401,29 @@ func (kub *Kubectl) DaemonSetIsReady(namespace, daemonset string) (int, error) {
...
+		return nil
+	}
+	up := strings.SplitN(cred, ":", 2)
+	if len(up) != 2 {
+		return fmt.Errorf("registry credentials had an invalid format")
+	}
+
+	cmd := fmt.Sprintf("%s secret docker-registry regcred --docker-server=%s --docker-username=%s --docker-password=%s", KubectlCmd, registry, up[0], up[1])
+	res = kub.ExecShort(cmd)
+	if !res.WasSuccessful() {
```
Comment (christarazi) [inline test/helpers/kubectl.go]: Nit:
```suggestion
	if !kub.ExecShort(cmd).WasSuccessful() {
```

File: test/helpers/kubectl.go [hunk 4]
```diff
@@ -401,6 +401,29 @@ func (kub *Kubectl) DaemonSetIsReady(namespace, daemonset string) (int, error) {
...
 }
 
+// AddRegistryCredentials adds a registry credentials secret into the
+// cluster
+func (kub *Kubectl) AddRegistryCredentials(cred string, registry string) error {
+	if len(cred) == 0 || cred == ":" {
+		return nil
+	}
+	res := kub.ExecShort(fmt.Sprintf("%s get secret regcred", KubectlCmd))
+	if res.WasSuccessful() {
```
Comment (christarazi) [inline test/helpers/kubectl.go]: Nit:
```suggestion
	if kub.ExecShort(fmt.Sprintf("%s get secret regcred", KubectlCmd)).WasSuccessful() {
```

**Review by christarazi — APPROVED** [review]
LGTM, minor nits

**Comment by christarazi** [comment]
Also, I believe backporting to older branches will have a slight gotcha where the Helm option for `imagePullSecrets` will presumably be under `global`?

**Comment by nathanjsweet** [comment]
@christarazi good catch. I'll make sure to do the backport myself, once this gets merged.

File: test/config/config.go (hunk 2 above)
Comment (nathanjsweet) [reply test/config/config.go:99]: All the other args are camel case...
cf: https://github.com/cilium/cilium/blob/7b846fd64a541b30b960f05efde454f237e2b948/test/config/config.go#L69

File: test/helpers/kubectl.go (hunk 3 above)
Comment (nathanjsweet) [reply test/helpers/kubectl.go]: I don't disagree with this sentiment, ~~but the rest of the file has the extra variable. I think we should keep the coding style consistent~~.

**Edit:**

Actually, you're right. Given that the variable is never used again I think it's justifiable to do this.

**Comment by nathanjsweet** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by pchaigno** [comment]
retest-net-next

**Review by tklauser — CHANGES_REQUESTED** [review]
One small nit on the commit message format: could you please add an empty line after the commit's subject? i.e.

```
    Add Registry Credentials to Tests

    In order to get around image registry pull limits,
    credentials can be set.
```
This will make it easier to backport the commit, as otherwise our `git grep` magic in the backport scripts doesn't seem to find commits formatted like this.

**Comment by nathanjsweet** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]

**Review by nebril — APPROVED** [review]

File: test/config/config.go (hunk 2 above)
Comment (christarazi) [reply test/config/config.go:99]: Ah, for some reason I only remembered `operator-image`, `operator-tag`, etc.

**Pull request merged**